packages/manager/apps/web/client/app/domain/anycast/domain-dns-anycast.controller.js

```diff
@@ -1,79 +1,91 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import {
+  DNS_ANYCAST_SERVICE_TYPE,
+  DNS_ANYCAST_PLANCODE,
+  DNS_ANYCAST_PRICING_MODE,
+  DNS_ANYCAST_DURATION,
+} from './domain-dns-anycast.constants';
+
 export default class DomainDnsAnycastActivateCtrl {
   /* @ngInject */
-  constructor($translate, $window, Alerter, atInternet, Domain) {
-    this.$translate = $translate;
-    this.$window = $window;
-    this.Alerter = Alerter;
-    this.atInternet = atInternet;
-    this.Domain = Domain;
+  constructor($q, WucOrderCartService, SessionService) {
+    this.$q = $q;
+    this.WucOrderCartService = WucOrderCartService;
+    this.SessionService = SessionService;
   }
 
   $onInit() {
-    this.domainId = this.domainName;
-    this.optionName = 'dnsAnycast';
-    this.loading = false;
-    this.loadOptionDetails();
-  }
-
-  loadOptionDetails() {
     this.loading = true;
-    this.error = null;
-
-    return this.Domain.getOptionDetails(this.domainId, this.optionName)
-      .then((data) => {
-        this.optionDetails = data;
-      })
-      .catch(() => {
-        this.error = this.$translate.instant(
-          'domain_configuration_dnsanycast_fail',
+    return this.fetchCatalogOffer(this.domainName)
+      .then((offer) => {
+        this.offer = offer;
+        this.offerPrice = find(this.offer.prices, {
+          pricingMode: DNS_ANYCAST_PRICING_MODE,
+          duration: DNS_ANYCAST_DURATION,
+        });
+        return this.fetchNewCart().then((cart) =>
+          this.WucOrderCartService.addProductServiceOptionToCart(
+            cart.cartId,
+            DNS_ANYCAST_SERVICE_TYPE,
+            this.domainName,
+            {
+              duration: this.offerPrice.duration,
+              planCode: this.offer.planCode,
+              pricingMode: this.offerPrice.pricingMode,
+              quantity: this.offerPrice.minimumQuantity,
+            },
+          )
+            .then(() =>
+              this.WucOrderCartService.getCheckoutInformations(cart.cartId),
+            )
+            .then((checkout) => {
+              this.cart = cart;
+              this.checkoutInfos = checkout;
+            }),
         );
       })
+      .catch((error) => {
+        this.error = get(error, 'data.message', error);
+      })
       .finally(() => {
         this.loading = false;
       });
   }
 
-  orderDnsanycast() {
-    this.url = null;
-    this.error = null;
-    this.loading = true;
+  fetchNewCart() {
+    return this.SessionService.getUser().then(({ ovhSubsidiary }) =>
```

Comment:
Maybe we could inject use the `user` resolve injected in the app root state instead

Reply:
done